compiler/AST/AggregateType.cpp

```diff
@@ -194,9 +194,21 @@ DefExpr* AggregateType::toSuperField(SymExpr*  expr) {
 
   if (isClass() == true) {
     forv_Vec(AggregateType, pt, dispatchParents) {
+      AggregateType* root = pt->getRootInstantiation();
       if (DefExpr* field = pt->toLocalField(expr)) {
         retval = field;
         break;
+      } else if (DefExpr* field = pt->toSuperField(expr)) {
+        retval = field;
+        break;
+      } else if (pt != root) {
+        if (DefExpr* field = root->toLocalField(expr)) {
```

Comment:
We create new field symbols when creating an instantiation, and this case is needed because SymExprs in the initializer continue to point to the generic class.

Reply:
There's a helper function that I've used in the past to whole-sale replace old references with their updated versions.  I believe you want to use that instead.  Let me try to track it down, just a sec.

Reply:
It's called `update_symbols`, and there's an example use of it [here](https://github.com/chapel-lang/chapel/blob/master/compiler/passes/initializerRules.cpp#L1088).  There are probably some uses of it in the generic instantiation for types with initializers, or at least there should be

Reply:
I originally tried that with the 'genericFieldMap', which this PR replaces. Something like genericFieldMap could probably work eventually, but I thought it best to try and stick to the previously-established 'updateFieldsMember' approach. I intend to make a cleanup pass across this chain of functions (toLocalField, toSuperField, updateFieldsMember) at a later time.

Reply:
Ah, okay.  I still think it is strange that the generic instantiation generation didn't handle this already, but I'm not going to block the PR on it.  After the error message update, this should be good to go!